pkg/cmd/cli/install/install.go

```diff
@@ -0,0 +1,187 @@
+func NewCommand(f client.Factory) *cobra.Command {
+	o := NewInstallOptions()
+	c := &cobra.Command{
+		Use:   "install",
+		Short: "Install Velero",
+		Long: `
+Install Velero onto a Kubernetes cluster using the supplied provider information, such as
+the provider's name, a bucket name, and a file containing the credentials to access that bucket.
+A prefix within the bucket and configuration for the backup store location may also be supplied.
+Additionally, volume snapshot information for the same provider may be supplied.
+
+All required CustomResourceDefinitions will be installed to the server, as well as the
+Velero Deployment and associated Restic DaemonSet.
+
+The provided secret data will be created in a Secret named 'cloud-credentials'.
+
+All namespaced resources will be placed in the 'velero' namespace.
+
+Use '-o yaml' or '-o json'  with '--dry-run' to output all generated resources as text instead of sending the resources to the server.
+This is useful as a starting point for more customized installations.
+		`,
+		Run: func(c *cobra.Command, args []string) {
+			o.Namespace = c.Flag("namespace").Value.String()
+			cmd.CheckError(o.Validate(c, args, f))
+			cmd.CheckError(o.Complete(args, f))
+			cmd.CheckError(o.Run(c))
+		},
+	}
+
+	o.BindFlags(c.Flags())
+	output.BindFlags(c.Flags())
+	output.ClearOutputFlagDefault(c)
+
+	return c
+}
+
+// Run executes a command in the context of the provided arguments.
+func (o *InstallOptions) Run(c *cobra.Command) error {
+	realPath, err := filepath.Abs(o.Secret)
+	if err != nil {
+		return err
+	}
+	secretData, err := ioutil.ReadFile(realPath)
+	if err != nil {
+		return err
+	}
+	resources, err := install.AllResources(
+		o.Namespace,
+		o.Image,
+		o.ProviderName,
+		o.BucketName,
+		o.Prefix,
+		o.BackupStorageConfig.Data(),
+		o.VolumeSnapshotConfig.Data(),
+		secretData,
+		o.RestoreOnly,
+	)
+	if err != nil {
+		return err
+	}
+
+	if _, err := output.PrintWithFormat(c, resources); err != nil {
+		return err
+	}
+
+	if o.DryRun {
+		return nil
+	}
+
+	clientConfig, err := client.Config("", "", fmt.Sprintf("%s-%s", c.Parent().Name(), c.Name()))
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := dynamic.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	err = install.Install(client.NewDynamicFactory(dynamicClient), resources, os.Stdout)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//Complete completes options for a command.
+func (o *InstallOptions) Complete(args []string, f client.Factory) error {
+	return nil
+}
+
+// Validate validates options provided to a command.
+func (o *InstallOptions) Validate(c *cobra.Command, args []string, f client.Factory) error {
+	if err := output.ValidateFlags(c); err != nil {
+		return err
+	}
+
+	if o.BucketName == "" {
+		return errors.New("--bucket is required")
+	}
+
+	if o.ProviderName == "" {
+		return errors.New("--provider is required")
+	}
+
+	if o.Secret == "" {
+		return errors.New("--secret is required")
+	}
+
```

Comment:
probably makes sense to add a check here that the value of the `-o` flag is not `table`.

Reply:
I actually put this further down the call chain, since we use `output.ValidateFlags`. I'll see if I can move it into that function.